File: short-api/src/main/java/com/sidalifetoumi/shorter/url/Url.java
package com.sidalifetoumi.shorter.url;

/*
    @Author Fetoumi Sidali
    @Created 11/23/2025
*/

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document(collection = "urls")
public class Url {

    @Id
    private String id;

    private String longUrl;

    private String shortId;

    @CreatedDate
    private LocalDateTime createdAt;
}

File: short-ui/src/types/url.ts
export interface CreateUrlRequest{
    url:string
}
export interface ShortUrlResponse {
    shortUrl:string
}

export interface Url {
    id:string
    originalUrl:string,
    shortUrl:string
}

export interface DatabaseUrl{
    id:string,
    longUrl:string,
    shortId:string,
    createdAt: string
}